librarian_server/orm/sendqueue.py

```diff
@@ -0,0 +1,183 @@
+simply copied and the server waits for the copy to complete. In the latter
+scenario, the server places an item in this priority queue and a separate
+thread handles the transfer.
+
+There are a few reasons for this:
+
+- Reducing load on the server and background task threads.
+- Allowing for batching of transfers.
+- Ability to track retries and failures.
+"""
+
+import datetime
+from typing import TYPE_CHECKING
+
+from sqlalchemy.orm import Session
+
+from hera_librarian.async_transfers import CoreAsyncTransferManager
+from hera_librarian.errors import ErrorCategory, ErrorSeverity
+from hera_librarian.exceptions import LibrarianError
+from hera_librarian.models.checkin import CheckinUpdateRequest, CheckinUpdateResponse
+from hera_librarian.transfer import TransferStatus
+
+from .. import database as db
+from ..logger import log, log_to_database
+from .librarian import Librarian
+
+if TYPE_CHECKING:
+    from .transfer import OutgoingTransfer
+
+
+class SendQueue(db.Base):
+    """
+    The priority queue for sending data to other stores, librarians, and clinets
+    using a simple local copy.
+    """
+
+    __tablename__ = "send_queue"
+
+    id = db.Column(db.Integer, primary_key=True)
+    "The unique identifier for this item in the queue."
+    priority = db.Column(db.Integer, nullable=False)
+    "The priority of this item in the queue. Lower is higher priority."
+    created_time: datetime.datetime = db.Column(db.DateTime, nullable=False)
+    "The time that this item was added to the queue."
+    transfers = db.relationship("OutgoingTransfer", back_populates="send_queue")
+    "The transfers this item is associated with."
+    retries = db.Column(db.Integer, nullable=False)
+    "The number of times this item has been tried to be sent."
+
+    destination = db.Column(db.String(256), nullable=False)
+    "The name of the destination librarian."
+
+    async_transfer_manager = db.Column(db.PickleType)
+    "The transfer manager that this queue item will use."
+
+    consumed: bool = db.Column(db.Boolean, default=False)
+    "Whether this queue item has been consumed."
+    consumed_time: datetime = db.Column(db.DateTime)
+    "The time at which the item was consumed. Useful for pruning."
+
+    completed: bool = db.Column(db.Boolean, default=False)
+    "Whether this queue item has been entirely completed."
+    completed_time: datetime = db.Column(db.DateTime)
+    "The time at which the queue item was marked as completed."
+
+    @classmethod
+    def new_item(
+        cls,
+        priority: int,
+        destination: str,
+        transfers: list["OutgoingTransfer"],
+        async_transfer_manager: CoreAsyncTransferManager,
+    ) -> "SendQueue":
+        """
+        Create a new item in the queue.
+
+        Parameters
+        ----------
+        priority : int
+            The priority of this transfer. Older, and higher priority, transfers will be completed first.
+            Higher numbers mean higher priority.
```

Comment:
Line 49 "Lower is higher priority" here "higher is higher priority". Which one is the correct case?